cmd/app.go

```diff
@@ -1131,20 +1124,36 @@ func getAnswersFile(location string, answers map[string]string) error {
 	return nil
 }
 
-// getValuesFile reads a values file and parse it to answers in helm strvals format
-func getValuesFile(location string, answers map[string]string) error {
-	bytes, err := readFileReturnJSON(location)
+// parseValuesFile reads a values file and parses it to answers in helm strvals format
+func parseValuesFile(location string, answers map[string]string) error {
+	values, err := parseFile(location)
 	if err != nil {
 		return err
 	}
-	values := make(map[string]interface{})
-	if err := json.Unmarshal(bytes, &values); err != nil {
-		return err
-	}
 	valuesToAnswers(values, answers)
 	return nil
 }
 
+func parseFile(location string) (map[string]interface{}, error) {
+	bytes, err := ioutil.ReadFile(location)
+	if err != nil {
+		return nil, err
+	}
+
+	values := make(map[string]interface{})
+	if hasPrefix(bytes, []byte("{")) {
+		// this is the check that "readFileReturnJSON" uses to differentiate between JSON and YAML
```

Comment:
Can we get rid of the func `readFileReturnJSON` altogether from the repo?

Reply:
So it's still being used in getRKEConfig, didn't want to change that area with this pr

Reply:
You can leave it, it would require a bigger refactor since everything is working with bytes

Reply:
Based on offline discussion with @dramich and @leodotcloud , we'll keep `readFileReturnJSON` for `getRKEConfig`, since that method expects the output from `readFileReturnJSON` in bytes and calls a couple more methods on it before calling unmarshal. Making all those changes seems out of the scope of this pr, especially since we don't have a proper way of testing because currently creating cluster by passing rke-config only extracts top level keys.